packages/@aws-cdk/aws-elasticsearch/lib/elasticsearch-access-policy.ts

```diff
@@ -30,9 +30,7 @@ export interface ElasticsearchAccessPolicyProps {
  */
 export class ElasticsearchAccessPolicy extends cr.AwsCustomResource {
```

Comment:
I'm sorry if I was unclear in my previous review. This construct needs a private class property `policyStatements` and a method `addPolicyStatement` that pushes a statement onto this array. The call to `Lazy` then renders the policy document using the value of `policyStatement` (at synthesis time, since it's a `Lazy`).

Reply:
Moved the policy statements storage to this construct.